pkg/generators/forwarding/fluentd/fluent_conf_test.go

```diff
@@ -73,6 +74,60 @@ var _ = Describe("Generating fluentd config", func() {
 		}
 	})
 
+	It("should generats container source config for given namespaces only", func() {
+		forwarding = &logging.ForwardingSpec{
+			Outputs: []logging.OutputSpec{
+				{
+					Type:     logging.OutputTypeElasticsearch,
+					Name:     "apps-es-1",
+					Endpoint: "es.svc.messaging.cluster.local:9654",
+					Secret: &logging.OutputSecretSpec{
+						Name: "my-es-secret",
+					},
+				},
+			},
+			Pipelines: []logging.PipelineSpec{
+				{
+					Name:       "apps-pipeline",
+					SourceType: logging.LogSourceTypeApp,
+					OutputRefs: []string{"apps-es-1", "apps-es-2"},
+					Namespaces: []string{"project1-namespace", "project2-namespace"},
+				},
+			},
+		}
+		results, err := generator.generateSource(sets.NewString(string(forwarding.Pipelines[0].SourceType)), sets.NewString(forwarding.Pipelines[0].Namespaces...))
+		Expect(err).To(BeNil())
+		test.Expect(results[0]).ToEqual(`
+# container logs
+<source>
+  @type tail
+  @id container-input
+  path "/var/log/containers/*_project1-namespace_*.log", "/var/log/containers/*_project2-namespace_*.log"
+  exclude_path ["/var/log/containers/fluentd-*_openshift-logging_*.log", "/var/log/containers/elasticsearch-*_openshift-logging_*.log", "/var/log/containers/kibana-*_openshift-logging_*.log"]
+  pos_file "/var/log/es-containers.log.pos"
```

Comment:
According to the fluentd tail plugin [docs](https://docs.fluentd.org/input/tail#pos_file-highly-recommended) the `pos_file` should not be shared across `in_tail` configurations. Assuming multiple pipelines with independent Namespace filter, are we safe with a shared `pos_file` definition? (cc @jcantrill , @ewolinetz)

Reply:
There is only one in_tail configuration generated for multiple namespace filter. So no sharing of pos_file.